parsers/regex/regex.go

```diff
@@ -34,74 +33,110 @@ type Parser struct {
-			return errors.New(fmt.Sprintf("No named capture groups found in regex: '%s'. Must provide at least one named group with line regex. Example: `(?P<name>re)`", p.conf.LineRegex))
+	// Require at least one named group
+	var numNamedGroups int
+	for _, groupName := range lineRegex.SubexpNames() {
+		if groupName != "" {
+			numNamedGroups++
 		}
-
-		p.lineParser = &RegexLineParser{lineRegex}
+	}
+	if numNamedGroups == 0 {
+		logrus.WithFields(logrus.Fields{
+			"LineRegex": regexStr,
+		}).Error("No named capture groups")
+		return nil, errors.New(fmt.Sprintf("No named capture groups found in regex: '%s'. Must provide at least one named group with line regex. Example: `(?P<name>re)`", regexStr))
 	}
 
-	return nil
+	return lineRegex, nil
 }
 
+/* LineParser for regexes */
+
 type RegexLineParser struct {
-	lineRegex *regexp.Regexp
+	lineRegexes []*regexp.Regexp
 }
 
-func (p *RegexLineParser) ParseLine(line string) (map[string]interface{}, error) {
-	parsed := make(map[string]interface{})
-	if p.lineRegex == nil {
-		logrus.Error("RegexLineParser", p, "has no lineRegex")
-		return parsed, errors.New("RegexLineParser instance has no lineRegex")
-	}
-	match := p.lineRegex.FindAllStringSubmatch(line, -1)
-	if match == nil || len(match) == 0 {
-		logrus.WithFields(logrus.Fields{
-			"line": line,
-		}).Debug("No matches for regex log line")
-		return parsed, nil
+// RegexLineParser factory
+func NewRegexLineParser(regexStrs []string) (*RegexLineParser, error) {
+	lineRegexes, err := ParseLineRegexes(regexStrs)
+	fmt.Println("lineRegexes", lineRegexes)
+	if err != nil {
+		return nil, err
 	}
+	logrus.WithFields(logrus.Fields{
+		"lineRegexes": lineRegexes,
+	}).Debug("Compiled line regexes")
+	return &RegexLineParser{lineRegexes}, nil
+}
 
-	// Map capture groups
-	var firstMatch []string = match[0] // We only care about the first full lineRegex match
-	for i, name := range p.lineRegex.SubexpNames() {
-		if i != 0 && i < len(firstMatch) {
-			parsed[name] = firstMatch[i]
+func (p *RegexLineParser) ParseLine(line string) (map[string]interface{}, error) {
+	for _, lineRegex := range p.lineRegexes {
+		parsed := make(map[string]interface{})
+		match := lineRegex.FindAllStringSubmatch(line, -1)
+		if match == nil || len(match) == 0 {
+			logrus.WithFields(logrus.Fields{
+				"line": line,
+			}).Debug("No matches for regex log line")
+			continue // No matches found, skip to next regex
 		}
-	}
-	logrus.WithFields(logrus.Fields{
-		"parsed": parsed,
-		"line":   line,
-	}).Debug("Regex parsing log line")
 
-	return parsed, nil
+		// Map capture groups
+		var firstMatch []string = match[0] // We only care about the first full lineRegex match
+		for i, name := range lineRegex.SubexpNames() {
+			if i != 0 && i < len(firstMatch) {
+				parsed[name] = firstMatch[i]
+			}
+		}
+		logrus.WithFields(logrus.Fields{
+			"parsed":    parsed,
+			"line":      line,
+			"lineRegex": lineRegex,
+		}).Debug("Regex parsing log line")
+
+		return parsed, nil
```

Comment:
I'd like a comment somewhere (maybe the overall docs?) saying something like "first regex to match the line wins, so make sure to list your regexes in most-specific to least-specific order"